File: src/NetWorthCalculator.Web.ReactJs/src/Models/Asset/Asset.ts
import { observable } from "mobx";
import { serializable } from "serializr";
import { AssetGroup } from "./AssetGroup";

interface IAsset {
	id: number;
	amount: number;
	group: AssetGroup;
	description: string;
	order: number;
}

export default class Asset {

	@serializable @observable id!: number;

	@serializable @observable amount!: number;

	@serializable @observable group!: AssetGroup;

	@serializable @observable description!: string;

	@serializable @observable order!: number;

	constructor(values: IAsset) {
		Object.assign(this, {...values});
	}
}

File: src/NetWorthCalculator.Entities/Models/Asset.cs
using NetWorthCalculator.Entities.Enums;

namespace NetWorthCalculator.Entities.Models
{
    public class Asset
    {
        public int Id { get; set; }

        public decimal Amount { get; set; }

        public string Description { get; set; }

        public AssetGroup Group { get; set; }

        public int Order { get; set; }
    }
}
